Makefile

```diff
@@ -22,29 +17,34 @@ submodules:
 
 lint:
 	@echo "Running flake8"
-	@cd src && tox -e lint
+	@tox -e lint
 
 test: unittest functional lint
 
 unittest:
-	@cd src && tox -e unit
+	@tox -e unit
 
 functional: build
-	@cd src && tox -e functional
+	@tox -e functional
 
 build:
 	@echo "Building charm to base directory $(JUJU_REPOSITORY)"
-	@-git describe --tags > ./src/repo-info
-	@LAYER_PATH=./layers INTERFACE_PATH=./interfaces TERM=linux\
-		JUJU_REPOSITORY=$(JUJU_REPOSITORY) charm build ./src --force
+	@-git describe --tags > ./repo-info
+	@LAYER_PATH=./layers INTERFACE_PATH=./interfaces TERM=linux \
+		JUJU_REPOSITORY=$(JUJU_REPOSITORY) charm build . --force
 
 release: clean build
 	@echo "Charm is built at $(JUJU_REPOSITORY)/builds"
 
+push: release
+	@echo "Pushing $(JUJU_REPOSITORY)/builds/gitlab"
+	@charm push $(JUJU_REPOSITORY)/builds/gitlab cs:~pirate-charmers/gitlab
```

Comment:
You've left 'gitlab' hard coded this section needs to use template variables or be left for the charm to implement.

Reply:
Fixed